pkg/rollout/rollout.go

```diff
@@ -200,6 +208,41 @@ func (r *Rollout) PrepareRollback(svc *run.Service, stable, candidate string) *r
 	return svc
 }
 
+// updateServiceBasedOnDiagnosis updates a service's traffic configuration
+// based on the diagnosis.
+// If service was unchanged, nil is returned.
+//
+// TODO (gvso): Refactor this and other methods to return only a traffic object.
+// Then, rename this to trafficBasedOnDiagnosis.
+// Also, move those traffic-config-related methods to traffic.go file.
```

Comment:
Instead of updating the service in `PrepareRollForward` and `PrepareRollForward`, which are down in the call chains, we can just return the traffic config and make the update in, yeah, `UpdateService`.

We need to change the first two method's name and move them (an their helpers) to a new file since this one is getting very big

Reply:
I think the TODO above is a good idea.  Though I'm fine with it as it is right now.